src/Components/EditAgent.jsx

```diff
@@ -1,141 +1,110 @@
+        fetch(`${config.API_ENDPOINT}/agent/${agentToEdit.agentId}`, init)
+            .then(response => {
+                if (response.status === 204) {
+                    return null;
+                } else if (response.status === 400) {
+                    return response.json();
+                }
+                return Promise.reject('Something unexpected went wrong :)');
+            })
+            .then(data => {
+                if (!data) {
+                    // redirect the user back to the /agents route
+                    history.push('/agents');
+                } else {
+                    // we have errors to display
+                    setErrors(data);
+                }
+            })
+            .catch(error => console.log(error));
     };
 
+    if(!agentToEdit){
+        return null;
+    }
+
     return (
         <main>
             <div className="wrapper">
                 <div className="typing-demo-edit">
                     <h1>Edit Agent.</h1>
                 </div>
             </div>
+            <Errors errors={errors} />
             <BackButton/>
             <form onSubmit={editAgentFormSubmitHandler}>
                 <ul className="form-style-1">
                     <li>
-                        <label>Full Name <span className="required">*</span></label>
-                        <input type="text" name="field1" className="field-divided" placeholder={firstName} onChange={firstNameOnChangeHandler}/>
-                        <input type="text" name="field2" className="field-divided" placeholder={lastName} onChange={lastNameOnChangeHandler}/>
+                        <label>First Name <span className="required">*</span></label>
+                        <input type="text" name="firstName" className="field-divided" value={agentToEdit.firstName} onChange={editInputOnChangeHandler}/>
+                        <label>Last Name <span className="required">*</span></label>
+                        <input type="text" name="lastName" className="field-divided" value={agentToEdit.lastName} onChange={editInputOnChangeHandler}/>
                     </li>
                     <li>
                         <label htmlFor="date">Birth Date <span className="required">*</span></label>
-                        <input type="date" name="date" id="date" onFocus={`this.type=${dob}`} onBlur="(this.type='text')" onChange={dobOnChangeHandler}/>
+                        <input type="date" name="dob" id="date" value={agentToEdit.dob} onChange={editInputOnChangeHandler}/>
                     </li>
                     <li>
                         <label>Height (inches)<span className="required">*</span></label>
-                        <input type="text" name="field3" className="field-long" placeholder={height} onChange={heightOnChangeHandler}/>
+                        <input type="text" name="height" className="field-long" value={agentToEdit.heightInInches} onChange={editInputOnChangeHandler}/>
```

Comment:
There is a bug when editing height. The field won't change. Haven't found it yet...

Reply:
Found it...